persistence/sql/persister_oauth2.go

```diff
@@ -280,13 +350,27 @@ func (p *Persister) deactivateSessionByRequestID(ctx context.Context, id string,
 	return sqlcon.HandleError(
 		p.Connection(ctx).
 			RawQuery(
-				fmt.Sprintf("UPDATE %s SET active=false WHERE request_id=?", OAuth2RequestSQL{Table: table}.TableName()),
+				fmt.Sprintf("UPDATE %s SET active=false, used=true WHERE request_id=?", OAuth2RequestSQL{Table: table}.TableName()),
 				id,
 			).
 			Exec(),
 	)
 }
 
+func (p *Persister) getRefreshTokenUsedStatusBySignature(ctx context.Context, signature string) (bool, error) {
```

Comment:
Please add tests for this new method

Reply:
this is a private method, do you have a suggestion on how I can test it ? Alternatively testing `RevokeRefreshTokenMaybeGracePeriod` will test this method indirectly.

Reply:
You can export it if need be :)